components/scream/src/physics/shoc/shoc_diag_second_moments_lbycond_impl.hpp

```diff
@@ -6,20 +6,42 @@
 namespace scream {
 namespace shoc {
 
-/*
- * Implementation of shoc diag_second_moments_lbycond. Clients should NOT
- * #include this file, but include shoc_functions.hpp instead.
- */
-
 template<typename S, typename D>
 KOKKOS_FUNCTION
-void Functions<S,D>::diag_second_moments_lbycond(const Int& shcol, const uview_1d<const Spack>& wthl_sfc, const uview_1d<const Spack>& wqw_sfc, const uview_1d<const Spack>& uw_sfc, const uview_1d<const Spack>& vw_sfc, const uview_1d<const Spack>& ustar2, const uview_1d<const Spack>& wstar, const uview_1d<Spack>& wthl_sec, const uview_1d<Spack>& wqw_sec, const uview_1d<Spack>& uw_sec, const uview_1d<Spack>& vw_sec, const uview_1d<Spack>& wtke_sec, const uview_1d<Spack>& thl_sec, const uview_1d<Spack>& qw_sec, const uview_1d<Spack>& qwthl_sec)
+void Functions<S,D>
+::shoc_diag_second_moments_lbycond(
+    const Scalar& wthl_sfc, const Scalar& wqw_sfc, const Scalar& uw_sfc, const Scalar& vw_sfc,
+    const Scalar& ustar2, const Scalar& wstar,
+    Scalar& wthl_sec, Scalar& wqw_sec, Scalar& uw_sec, Scalar& vw_sec,
+    Scalar& wtke_sec, Scalar& thl_sec, Scalar& qw_sec, Scalar& qwthl_sec)
 {
-  // TODO
-  // Note, argument types may need tweaking. Generator is not always able to tell what needs to be packed
-}
+  // Purpose of this subroutine is to diagnose the lower
+  // boundary condition for the second order moments needed
+  // for the SHOC parameterization.
+  // The thermodymnamic, tracer, and momentum fluxes are set
+  // to the surface fluxes for the host model, while the
+  // thermodynamic variances and covariances are computed
+  // according to that of Andre et al. 1978.
+
+  const auto ufmin = 0.01;
+
+  auto uf = std::sqrt(ustar2+sp(0.3)*wstar*wstar);
+  uf = ekat::impl::max<Scalar>(ufmin,uf);
 
+  // Diagnose thermodynamics variances and covariances
+  thl_sec   = sp(0.4)*sp(1.8)*((wthl_sfc/uf)*(wthl_sfc/uf));
+  qw_sec    = sp(0.4)*sp(1.8)*((wqw_sfc/uf)*(wqw_sfc/uf)); 
+  qwthl_sec = sp(0.2)*sp(1.8)*(wthl_sfc/uf)*(wqw_sfc/uf);
+
+  // Vertical fluxes of heat and moisture, simply
+  // use the surface fluxes given by host model
+  wthl_sec = wthl_sfc;
+  wqw_sec  = wqw_sfc;
+  uw_sec   = uw_sfc;
+  vw_sec   = vw_sfc;
+
+  wtke_sec = ekat::impl::max<Scalar>(sqrt(ustar2),ufmin)*ekat::impl::max<Scalar>(sqrt(ustar2),ufmin)*ekat::impl::max<Scalar>(sqrt(ustar2),ufmin);
```

Comment:
You can simplify this with `cube` unless that doesn't work here for some reason.

Reply:
cube function doesn't support Scalar argument at the moment.

Reply:
Would it be cleaner to do
```
max_val = ekat::impl::max(sqrt(ustar2),ufmin);
wtke_sec = max_val*max_val*max_val;
```
Just a thought, feel free to disregard if you prefer otherwise.

Reply:
I followed the fortran code in C++, not sure whether we want to make these changes. I guess it depends how people likes

Reply:
I recommend what @tcclevenger said